source/extensions/common/aws/sigv4a_key_derivation.cc

```diff
@@ -0,0 +1,148 @@
+#include "source/extensions/common/aws/sigv4a_key_derivation.h"
```

Comment:
It doesn't thrill me that we have code like this which I assume was copied from somewhere else per the below comments. Can we possibly bring that code in as a library in the future?

Reply:
The signing implementation is custom for Envoy, however the constant time functions were reproduced. It would be possible to include the standard AWS C Runtime as a library, but the dependency list is long (aws-c-auth, aws-lc, aws-c-cal, aws-c-common, aws-c-compression, aws-c-http, aws-c-io). It may be compatible with boringssl, but I've yet to test that and it is natively linked against s2n.

If we were able to agree on a path to linking against the C runtime, then I could look to PR that in future.

Reply:
I don't think we have the appetite for pulling all of that in. If there is some path in the future for putting the crypto into a dedicated library that would be easier to consume that would be nice but not a big deal.